diff -Naur refind-0.13.2-orig/Make.common refind-0.13.2/Make.common
--- refind-0.13.2-orig/Make.common	2022-01-08 07:29:18.574627281 -0800
+++ refind-0.13.2/Make.common	2022-01-08 07:29:34.337962198 -0800
@@ -35,12 +35,12 @@
 GENFW           = $(TIANOBASE)/BaseTools/Source/C/bin/GenFw
 prefix          = /usr/bin/
 ifeq ($(ARCH),aarch64)
-  CC            = $(prefix)aarch64-linux-gnu-gcc
-  AS            = $(prefix)aarch64-linux-gnu-as
-  LD            = $(prefix)aarch64-linux-gnu-ld
-  AR            = $(prefix)aarch64-linux-gnu-ar
-  RANLIB        = $(prefix)aarch64-linux-gnu-ranlib
-  OBJCOPY       = $(prefix)aarch64-linux-gnu-objcopy
+  CC            = $(prefix)aarch64-alpine-linux-musl-gcc
+  AS            = $(prefix)aarch64-alpine-linux-musl-as
+  LD            = $(prefix)ld
+  AR            = $(prefix)aarch64-alpine-linux-musl-gcc-ar
+  RANLIB        = $(prefix)aarch64-alpine-linux-musl-gcc-ranlib
+  OBJCOPY       = $(prefix)objcopy
 else
   CC            = $(prefix)gcc
   AS            = $(prefix)as
