diff -Naur ./Clients/dns-sd.c ./Clients/dns-sd.c
--- ./Clients/dns-sd.c	2021-01-26 12:47:47.000000000 -0800
+++ ./Clients/dns-sd.c	2022-02-06 10:20:14.250191773 -0800
@@ -57,6 +57,12 @@
 // aren't in the system's /usr/lib/libSystem.dylib.
 //#define TEST_NEW_CLIENTSTUB 1
 
+#define MIN(a,b) ((a)<(b)?(a):(b))
+#define va_start(v,l)   __builtin_va_start(v,l)
+#define va_end(v)       __builtin_va_end(v)
+#define va_arg(v,l)     __builtin_va_arg(v,l)
+#define va_copy(d,s)    __builtin_va_copy(d,s)
+
 #include <ctype.h>
 #include <stdio.h>          // For stdout, stderr
 #include <stdlib.h>         // For exit()
diff -Naur ./mDNSPosix/nss_mdns.c ./mDNSPosix/nss_mdns.c
--- ./mDNSPosix/nss_mdns.c	2020-07-21 16:17:17.000000000 -0700
+++ ./mDNSPosix/nss_mdns.c	2022-02-06 10:21:52.456868307 -0800
@@ -379,14 +379,20 @@
 #define ENTNAME  hostent
 #define DATABASE "hosts"
 
-#include <nss.h>
+//#include <nss.h>
 // For nss_status
 #include <netdb.h>
 // For hostent
 #include <sys/types.h>
 // For size_t
 
-typedef enum nss_status nss_status;
+#define NSS_STATUS_SUCCESS 0
+#define NSS_STATUS_NOTFOUND 1
+#define NSS_STATUS_UNAVAIL 2
+#define NSS_STATUS_TRYAGAIN 3
+#define NSS_STATUS_RETURN 4
+#define NETDB_INTERNAL NULL
+typedef int nss_status;
 typedef struct hostent hostent;
 
 /*
diff -Naur ./mDNSShared/CommonServices.h ./mDNSShared/CommonServices.h
--- ./mDNSShared/CommonServices.h	2020-07-21 16:17:17.000000000 -0700
+++ ./mDNSShared/CommonServices.h	2022-02-06 10:21:53.846868447 -0800
@@ -191,6 +191,7 @@
 
     #include    <stdint.h>
     #include    <arpa/inet.h>
+    #include    <sys/socket.h>
 
 #elif ( TARGET_OS_SOLARIS )
 
