--- ./mDNSPosix/Makefile	2021-01-26 12:47:47.000000000 -0800
+++ ./mDNSPosix/Makefile	2022-07-03 00:29:43.425837733 -0700
@@ -68,9 +68,9 @@
 ST = strip
 LD = ld
 SOOPTS = -shared
-CP = cp
+CP = install -D
 RM = rm
-LN = ln -s -f
+LN = ln -rs
 CFLAGS_COMMON = -I$(COREDIR) -I$(SHAREDDIR) -I$(DSODIR) -I$(PROXYDIR) -I$(OBJDIR) -fwrapv -W -Wall -DPID_FILE=\"/var/run/mdnsd.pid\" -DMDNS_UDS_SERVERPATH=\"/var/run/mdnsd\"
 CFLAGS_PTHREAD =
 LINKOPTS =
@@ -194,13 +194,13 @@
 NSSVERSION  := 0.2
 NSSLIBFILE  := $(NSSLIBNAME)-$(NSSVERSION).so
 NSSLINKNAME := $(NSSLIBNAME).so.2
-NSSINSTPATH := /lib
+NSSINSTPATH := $(DESTDIR)/usr/lib
 
 # If not otherwise defined, we install into /usr/lib and /usr/include
 # and our startup script is called mdns (e.g. /etc/init.d/mdns)
 ETCBASE?=/etc
-INSTBASE?=/usr
-STARTUPSCRIPTNAME?=mdns
+INSTBASE?=$(DESTDIR)/usr
+STARTUPSCRIPTNAME?=mdnsd
 
 ifeq ($(HAVE_IPV6),1)
 CFLAGS_OS += -DHAVE_IPV6=1
@@ -212,9 +212,9 @@
 
 # If directory /usr/share/man exists, then we install man pages into that, else /usr/man
 ifeq ($(wildcard /usr/share/man), /usr/share/man)
-MANPATH := /usr/share/man
+MANPATH := $(DESTDIR)/usr/share/man
 else
-MANPATH := /usr/man
+MANPATH := $(DESTDIR)/usr/man
 endif
 
 # If directories /etc/init.d/rc*.d exist, then we install into that (Suse)
@@ -233,7 +233,7 @@
 RUNLEVELSCRIPTSDIR = /etc
 else
 # Else install into /etc/rc.d/ (*BSD)
-STARTUPSCRIPTDIR = $(INSTBASE)/etc/rc.d
+STARTUPSCRIPTDIR = $(DESTDIR)/etc/rc.d
 endif
 endif
 endif
@@ -244,7 +244,7 @@
 
 all: setup Daemon libdns_sd Clients SAClient SAResponder SAProxyResponder NetMonitor $(OPTIONALTARG)
 
-install: setup InstalledStartup InstalledDaemon InstalledLib InstalledManPages InstalledClients $(OPTINSTALL)
+install: setup InstalledDaemon InstalledLib InstalledManPages InstalledClients $(OPTINSTALL)
 
 # 'setup' sets up the build directory structure the way we want
 setup:
@@ -311,8 +311,8 @@
 InstalledLib: $(INSTBASE)/lib/libdns_sd.$(LDSUFFIX).$(LIBVERS) $(INSTBASE)/include/dns_sd.h
 	@echo $+ " installed"
 
-InstalledStartup: $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME)
-	@echo $+ " installed"
+#InstalledStartup: $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME)
+#	@echo $+ " installed"
 
 InstalledManPages: $(MANPATH)/man8/mdnsd.8
 	@echo $+ " installed"
@@ -320,14 +320,14 @@
 InstalledClients: $(INSTBASE)/bin/dns-sd
 	@echo $+ " installed"
 
-InstalledNSS: $(NSSINSTPATH)/$(NSSLINKNAME) /etc/nss_mdns.conf $(MANPATH)/man5/nss_mdns.conf.5 $(MANPATH)/man8/libnss_mdns.8
+InstalledNSS: $(NSSINSTPATH)/$(NSSLINKNAME) $(DESTDIR)/etc/nss_mdns.conf $(MANPATH)/man5/nss_mdns.conf.5 $(MANPATH)/man8/libnss_mdns.8
 	@echo $+ " installed"
 
 # Note: If daemon already installed, we make sure it's stopped before overwriting it
-$(INSTBASE)/sbin/mdnsd: $(BUILDDIR)/mdnsd $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME)
-	if test -x $@; then $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME) stop; fi
-	$(CP) $< $@
-	$(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME) start
+$(INSTBASE)/sbin/mdnsd: $(BUILDDIR)/mdnsd 
+#	if test -x $@; then $(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME) stop; fi
+#	$(CP) $< $@
+#	$(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME) start
 
 $(INSTBASE)/lib/libdns_sd.$(LDSUFFIX).$(LIBVERS): $(BUILDDIR)/libdns_sd.$(LDSUFFIX)
 	$(CP) $< $@
@@ -340,32 +340,32 @@
 $(INSTBASE)/include/dns_sd.h: $(SHAREDDIR)/dns_sd.h
 	$(CP) $< $@
 
-$(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME): mdnsd.sh $(STARTUPSCRIPTDIR)
-	$(CP) $< $@
-	chmod ugo+x $@
-ifdef RUNLEVELSCRIPTSDIR
-ifeq ($(wildcard $(RUNLEVELSCRIPTSDIR)/runlevels/default), $(RUNLEVELSCRIPTSDIR)/runlevels/default)
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/runlevels/default/mdns
-else
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc2.d/S52mdns
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc3.d/S52mdns
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc4.d/S52mdns
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc5.d/S52mdns
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc0.d/K16mdns
-	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc6.d/K16mdns
-endif
-endif
+#$(STARTUPSCRIPTDIR)/$(STARTUPSCRIPTNAME): mdnsd.sh $(STARTUPSCRIPTDIR)
+#	$(CP) $< $@
+#	chmod ugo+x $@
+#ifdef RUNLEVELSCRIPTSDIR
+#ifeq ($(wildcard $(RUNLEVELSCRIPTSDIR)/runlevels/default), $(RUNLEVELSCRIPTSDIR)/runlevels/default)
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/runlevels/default/mdns
+#else
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc2.d/S52mdns
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc3.d/S52mdns
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc4.d/S52mdns
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc5.d/S52mdns
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc0.d/K16mdns
+#	$(LN) $@ $(RUNLEVELSCRIPTSDIR)/rc6.d/K16mdns
+#endif
+#endif
 
 $(MANPATH)/man5/%.5: %.5
-	cp $< $@
+	$(CP) $< $@
 	chmod 444 $@
 
 $(MANPATH)/man8/%.8: %.8
-	cp $< $@
+	$(CP) $< $@
 	chmod 444 $@
 
 $(MANPATH)/man8/mdnsd.8: $(SHAREDDIR)/mDNSResponder.8
-	cp $< $@
+	$(CP) $< $@
 	chmod 444 $@
 
 $(INSTBASE)/bin/dns-sd: ../Clients/build/dns-sd
@@ -373,19 +373,19 @@
 
 $(NSSINSTPATH)/$(NSSLINKNAME): $(NSSINSTPATH)/$(NSSLIBFILE)
 	$(LN) $< $@
-	ldconfig
+	#ldconfig
 
 $(NSSINSTPATH)/$(NSSLIBFILE): $(BUILDDIR)/$(NSSLIBFILE)
 	$(CP) $< $@
 	chmod 444 $@
 
-/etc/nss_mdns.conf: nss_mdns.conf
+$(DESTDIR)/etc/nss_mdns.conf: nss_mdns.conf
 	$(CP) $< $@
 	chmod 444 $@
 # Check the nsswitch.conf file.
 # If 'mdns' does not already appear on the "hosts:" line, then add it right before 'dns'
-	cp -f /etc/nsswitch.conf /etc/nsswitch.conf.pre-mdns
-	sed -e '/mdns/!s/^\(hosts:.*\)dns\(.*\)/\1mdns dns\2/' /etc/nsswitch.conf.pre-mdns > /etc/nsswitch.conf
+	#cp -f /etc/nsswitch.conf /etc/nsswitch.conf.pre-mdns
+	#sed -e '/mdns/!s/^\(hosts:.*\)dns\(.*\)/\1mdns dns\2/' /etc/nsswitch.conf.pre-mdns > /etc/nsswitch.conf
 
 #############################################################################
 
@@ -543,3 +543,6 @@
 $(OBJDIR)/%.l.o: $(SHAREDDIR)/%.l
 	$(FLEX) $(FLEXFLAGS_OS) -i             -o$(OBJDIR)/$*.l.c $<
 	$(CC) $(MDNSCFLAGS) -Wno-error -c -o $@ $(OBJDIR)/$*.l.c
+
+$(STARTUPSCRIPTDIR): $(DESTDIR)
+	mkdir -p $<
