diff -Naur openrc-0.44.9-orig/init.d/agetty.in openrc-0.44.9/init.d/agetty.in
--- openrc-0.44.9-orig/init.d/agetty.in	2022-08-18 21:57:21.290504467 +0000
+++ openrc-0.44.9/init.d/agetty.in	2022-08-18 21:57:41.795254683 +0000
@@ -14,7 +14,7 @@
 port="${RC_SVCNAME#*.}"
 respawn_period="${respawn_period:-60}"
 term_type="${term_type:-linux}"
-command=/sbin/agetty
+command=/usr/bin/agetty
 command_args_foreground="${agetty_options} ${port} ${baud} ${term_type}"
 pidfile="/run/${RC_SVCNAME}.pid"
 
