diff -Naur mkinitfs-4.0.0/Makefile mkinitfs-4.0.0-mod/Makefile
--- mkinitfs-4.0.0/Makefile	2022-07-26 21:19:05.000000000 +0000
+++ mkinitfs-4.0.0-mod/Makefile	2022-08-13 16:08:15.513110254 +0000
@@ -106,9 +106,11 @@
 LIBKMOD_LIBS	:= $(shell $(PKGCONF) --libs libkmod)
 CRYPTSETUP_CFLAGS := $(shell $(PKGCONF) --cflags libcryptsetup)
 CRYPTSETUP_LIBS	:= $(shell $(PKGCONF) --libs libcryptsetup)
+BSD_CFLAGS := $(shell $(PKGCONF) --cflags libbsd)
+BSD_LIBS := $(shell $(PKGCONF) --libs libbsd)
 
-CFLAGS		+= $(BLKID_CFLAGS) $(LIBKMOD_CFLAGS) $(CRYPTSETUP_CFLAGS)
-LIBS		= $(BLKID_LIBS) $(LIBKMOD_LIBS) $(CRYPTSETUP_LIBS)
+CFLAGS		+= $(BLKID_CFLAGS) $(LIBKMOD_CFLAGS) $(CRYPTSETUP_CFLAGS) $(BSD_CFLAGS)
+LIBS		= $(BLKID_LIBS) $(LIBKMOD_LIBS) $(CRYPTSETUP_LIBS) $(BSD_LIBS)
 
 %.o: %.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -o $@ -c $<
